{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "23cfc055",
   "metadata": {},
   "source": [
    "Notes:  drug-reactome.csv is mis-named and should be gene-reactome\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f43b7e59",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000228716\n",
      "http://purl.uniprot.org/uniprot/P40238\n",
      "all good!\n"
     ]
    }
   ],
   "source": [
    "require 'csv'\n",
    "\n",
    "# puts `pwd`\n",
    "# puts `ls -l`\n",
    "gene = {}\n",
    "CSV.foreach('./mappings/drug-gene-mappings.csv', headers: true) do |row|\n",
    "  gene[row[2]] = row[3] # 2 is the ENS-ID, 3 is the UniProt mapping\n",
    "end\n",
    "\n",
    "puts gene.first\n",
    "\n",
    "\n",
    "CSV.foreach('./rawdata/drug-reactome.csv', headers: true) do |row| # NOT DRUG - it is genes, actually\n",
    "  ensid = row[0]\n",
    "  abort \"not found #{row[0]}\" unless gene[ensid]\n",
    "end\n",
    "puts \"all good!\"  # we already know all of these genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eeb1e71e",
   "metadata": {},
   "source": [
    "# Process file\n",
    "\n",
    "using the existing mapping files rather than remote lookups\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07abb8ed",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "G alpha (s) signalling events\n",
      "Calcitonin-like ligand receptors\n",
      "ADORA2B mediated anti-inflammatory cytokines production\n",
      "ATP sensitive Potassium channels\n",
      "Regulation of insulin secretion\n",
      "Defective ABCC8 can cause hypo- and hyper-glycemias\n",
      "Signaling by ERBB4\n",
      "GABA receptor activation\n",
      "TNFs bind their physiological receptors\n",
      "Degradation of the extracellular matrix\n",
      "Assembly of collagen fibrils and other multimeric structures\n",
      "Anchoring fibril formation\n",
      "Laminin interactions\n",
      "Non-integrin membrane-ECM interactions\n",
      "Type I hemidesmosome assembly\n",
      "MET activates PTK2 signaling\n",
      "G alpha (s) signalling events\n",
      "ADORA2B mediated anti-inflammatory cytokines production\n",
      "Glucagon-type ligand receptors\n",
      "Respiratory electron transport\n",
      "Complex I biogenesis\n",
      "Nuclear signaling by ERBB4\n",
      "PIP3 activates AKT signaling\n",
      "Constitutive Signaling by Aberrant PI3K in Cancer\n",
      "Nuclear Receptor transcription pathway\n",
      "SUMOylation of intracellular receptors\n",
      "Ovarian tumor domain proteases\n",
      "PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling\n",
      "TFAP2 (AP-2) family regulates transcription of growth factors and their receptors\n",
      "RUNX1 regulates estrogen receptor mediated transcription\n",
      "ESR-mediated signaling\n",
      "RUNX1 regulates transcription of genes involved in WNT signaling\n",
      "Regulation of RUNX2 expression and activity\n",
      "Extra-nuclear estrogen signaling\n",
      "Estrogen-dependent gene expression\n",
      "NCAM1 interactions\n",
      "Translocation of SLC2A4 (GLUT4) to the plasma membrane\n",
      "Microtubule-dependent trafficking of connexons from Golgi to the plasma membrane\n",
      "Gap junction assembly\n",
      "MHC class II antigen presentation\n",
      "Separation of Sister Chromatids\n",
      "Resolution of Sister Chromatid Cohesion\n",
      "HSP90 chaperone cycle for steroid hormone receptors (SHR) in the presence of ligand\n",
      "Recruitment of NuMA to mitotic centrosomes\n",
      "Prefoldin mediated transfer of substrate  to CCT/TriC\n",
      "Formation of tubulin folding intermediates by CCT/TriC\n",
      "Post-chaperonin tubulin folding pathway\n",
      "Recycling pathway of L1\n",
      "Hedgehog 'off' state\n",
      "Cilium Assembly\n",
      "Intraflagellar transport\n",
      "RHO GTPases activate IQGAPs\n",
      "RHO GTPases Activate Formins\n",
      "COPI-mediated anterograde transport\n",
      "COPI-dependent Golgi-to-ER retrograde traffic\n",
      "COPI-independent Golgi-to-ER retrograde traffic\n",
      "Mitotic Prometaphase\n",
      "The role of GTSE1 in G2/M progression after G2 checkpoint\n",
      "Carboxyterminal post-translational modifications of tubulin\n",
      "HCMV Early Events\n",
      "Assembly and cell surface presentation of NMDA receptors\n",
      "Activation of AMPK downstream of NMDARs\n",
      "Aggrephagy\n",
      "EML4 and NUDC in mitotic spindle formation\n",
      "Sealing of the nuclear envelope (NE) by ESCRT-III\n",
      "Kinesins\n",
      "Neurophilin interactions with VEGF and VEGFR\n",
      "VEGF binds to VEGFR leading to receptor dimerization\n",
      "Recognition of DNA damage by PCNA-containing replication complex\n",
      "Polymerase switching on the C-strand of the telomere\n",
      "Processive synthesis on the C-strand of the telomere\n",
      "Telomere C-strand (Lagging Strand) Synthesis\n",
      "Removal of the Flap Intermediate from the C-strand\n",
      "Mismatch repair (MMR) directed by MSH2:MSH6 (MutSalpha)\n",
      "Mismatch repair (MMR) directed by MSH2:MSH3 (MutSbeta)\n",
      "PCNA-Dependent Long Patch Base Excision Repair\n",
      "Termination of translesion DNA synthesis\n",
      "HDR through Homologous Recombination (HRR)\n",
      "Gap-filling DNA repair synthesis and ligation in GG-NER\n",
      "Dual Incision in GG-NER\n",
      "Dual incision in TC-NER\n",
      "Gap-filling DNA repair synthesis and ligation in TC-NER\n",
      "Polymerase switching\n",
      "Removal of the Flap Intermediate\n",
      "Processive synthesis on the lagging strand\n",
      "Respiratory electron transport\n",
      "Complex I biogenesis\n",
      "Nuclear Receptor transcription pathway\n",
      "Carnitine shuttle\n",
      "Regulation of pyruvate dehydrogenase (PDH) complex\n",
      "Signaling by Retinoic Acid\n",
      "Interleukin-17 signaling\n",
      "Interleukin-4 and Interleukin-13 signaling\n",
      "Cholesterol biosynthesis\n",
      "PPARA activates gene expression\n",
      "Activation of gene expression by SREBF (SREBP)\n",
      "EGR2 and SOX10-mediated initiation of Schwann cell myelination\n",
      "Potential therapeutics for SARS\n",
      "G alpha (s) signalling events\n",
      "Cam-PDE 1 activation\n",
      "cGMP effects\n",
      "Tachykinin receptors bind tachykinins\n",
      "G alpha (q) signalling events\n",
      "Cargo recognition for clathrin-mediated endocytosis\n",
      "Clathrin-mediated endocytosis\n",
      "Thyroxine biosynthesis\n",
      "Highly sodium permeable postsynaptic acetylcholine nicotinic receptors\n",
      "Highly calcium permeable postsynaptic nicotinic acetylcholine receptors\n",
      "Highly calcium permeable nicotinic acetylcholine receptors\n",
      "Neutrophil degranulation\n",
      "Cell surface interactions at the vascular wall\n",
      "PIP3 activates AKT signaling\n",
      "Constitutive Signaling by Aberrant PI3K in Cancer\n",
      "PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling\n",
      "PI3K Cascade\n",
      "RAF/MAP kinase cascade\n",
      "FLT3 Signaling\n",
      "STAT5 Activation\n",
      "FLT3 mutants bind TKIs\n",
      "STAT5 activation downstream of FLT3 ITD mutants\n",
      "KW2449-resistant FLT3 mutants\n",
      "semaxanib-resistant FLT3 mutants\n",
      "crenolanib-resistant FLT3 mutants\n",
      "gilteritinib-resistant FLT3 mutants\n",
      "lestaurtinib-resistant FLT3 mutants\n",
      "midostaurin-resistant FLT3 mutants\n",
      "pexidartinib-resistant FLT3 mutants\n",
      "ponatinib-resistant FLT3 mutants\n",
      "quizartinib-resistant FLT3 mutants\n",
      "sorafenib-resistant FLT3 mutants\n",
      "sunitinib-resistant FLT3 mutants\n",
      "tandutinib-resistant FLT3 mutants\n",
      "linifanib-resistant FLT3 mutants\n",
      "tamatinib-resistant FLT3 mutants\n"
     ]
    }
   ],
   "source": [
    "require_relative '../Lookups/metadata_functions'\n",
    "require 'csv'\n",
    "\n",
    "f = File.open('./mappings/gene-reactome.csv', 'w')\n",
    "e = File.open('./mappings/gene-reactome-errors.txt', 'w')\n",
    "f.sync = true # Ensure immediate writes\n",
    "e.sync = true # Ensure immediate writes\n",
    "f.write \"radboudsource,sourceguid,target,targetguid,targetlabel\\n\"\n",
    "\n",
    "CSV.foreach('./rawdata/drug-reactome.csv', headers: true) do |row|  # not drug, gene!\n",
    "  \n",
    "  next if row.size < 3\n",
    "  radboudsource = row[0]\n",
    "  sourceguid = gene[radboudsource]\n",
    "  target = row[2]\n",
    "  targetguid = \"http://identifiers.org/reactome/#{target}\"    # this is the URI used by UniProt\n",
    "  r = REACTOME.new(uri: target)\n",
    "  targetlabel = r.lookup_title\n",
    "  warn targetlabel\n",
    "\n",
    "  f.write CSV.generate_line([radboudsource,sourceguid,target,targetguid,targetlabel])\n",
    "\n",
    "end\n",
    "\n",
    "f.close\n",
    "e.close\n",
    "puts \"done\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Ruby 3.0.0",
   "language": "ruby",
   "name": "ruby"
  },
  "language_info": {
   "file_extension": ".rb",
   "mimetype": "application/x-ruby",
   "name": "ruby",
   "version": "3.0.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
